[move][doc] phantom types

mengxu-fb opened the pull request:
Adds a section about phantom types to the Move book.

## Motivation

Docs are good

### Have you read the [Contributing Guidelines on pull requests](https://github.com/diem/diem/blob/main/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

`mdbook build`

tnowacki commented on language/documentation/book/src/generics.md:
```diff
@@ -135,6 +135,8 @@ module M {
 
 ## Unused Type Parameters
 
+An unused type parameter is one that doesn't show up at runtime,
```
I would reword this a bit "does not appear in a field" or something similar, as it does appear "at runtime" in the VM

tnowacki reviewed (commented):
For a general formatting thing, I would check out Prettier for keeping some of the lines to the same length. Not a big deal though

mengxu-fb commented:
Addressed comments in the `fixup!` commit.

A checked at Prettier and am wondering when would be the best time to apply it.

I can either apply it to the `generics.md` file only (and will only update this file). Or later at some time we can apply it to all the `.md` files in the Move Book folder and format them once. I am open to either way though.

mengxu-fb commented:
/land
